### PR TITLE
refactor(landmark_based_localizer): refactored landmark_tf_caster

### PR DESCRIPTION
**Pull request by SakodaShintaro**
## Description

* Removed `landmark_tf_caster` node, and modified the package as a library providing parsing functions.
  * Renamed `landmark_tf_caster` to `landmark_parser`
* Landmark information is not informed via `/tf_static`, but is held by each landmark_based_localizer as a `std::map<std::string, geometry_msgs::msg::Pose>`.
* Subscribe to `autoware_auto_mapping_msgs::msg::HADMapBin` is also performed by each landmark based localizer.
* Added debug topic `~/debug/marker` 
  * can be watched by  image
  * result 
Screenshot from 2023-10-26 14-39-16

## Tests performed

The AR tag based localizer is confirmed to work properly.

## Effects on system behavior

There is no change in the behavior on the system as a whole.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: localization/landmark_based_localizer/landmark_parser/package.xml [hunk 1]
```diff
@@ -1,22 +1,23 @@
...
   <maintainer email="masahiro.sakamoto@tier4.jp">Masahiro Sakamoto</maintainer>
+  <maintainer email="yamato.ando@tier4.jp">Yamato Ando</maintainer>
+  <maintainer email="kento.yabuuchi.2@tier4.jp">Kento Yabuuchi</maintainer>
   <license>Apache License 2.0</license>
 
   <buildtool_depend>ament_cmake</buildtool_depend>
   <buildtool_depend>autoware_cmake</buildtool_depend>
 
   <depend>autoware_auto_mapping_msgs</depend>
+  <depend>eigen</depend>
```
Comment (YamatoAndo) [inline localization/landmark_based_localizer/landmark_parser/package.xml:17]:
```suggestion
  <build_depend>eigen</build_depend>
```

**Review by YamatoAndo — APPROVED** [review]

**Comment by KYabuuchi** [comment]
@SakodaShintaro In my environment, some landmarks does not appear in rviz. Do you know what is causing it?
image

**Comment by KYabuuchi** [comment]
:arrow_up:  now I understand the cause.
I think it would be better to use `visualization_msgs::msg::MarkerArray` and publish it as one topic.

**Comment by SakodaShintaro** [comment]
@KYabuuchi 
I agree 100% that it would be better to use a single MarkerArray instead of multiple Markers. Fixed in https://github.com/autowarefoundation/autoware.universe/pull/5414/commits/acbd19e0d2d9eea97e0bcc198610afc6edc8909c

**Review by KYabuuchi — COMMENTED** [review]
LGTM :ok_hand:

**Review by KYabuuchi — APPROVED** [review]

**Review by Motsu-san — APPROVED** [review]
[LGTMeow](https://lgtmeow.com)

**Pull request merged**